### PR TITLE
Deprecate hierarchicalSelectors option

### PR DESCRIPTION
**Pull request by evilebottnawi**
https://github.com/stylelint/stylelint/issues/1309
/cc @davidtheclark @jeddy3 all right?

### CODE REVIEW

**Comment by jeddy3** [comment]
Lets get a release out with https://github.com/stylelint/stylelint/pull/1332 and https://github.com/stylelint/stylelint/pull/1314 before merging in this. So that we can release a new version of `config-standard` out without drip-feeding the deprecation warnings.

After that, I think the priority is https://github.com/stylelint/stylelint/issues/1316. I've updated the milestones accordingly.

File: src/rules/indentation/index.js [hunk 1]
```diff
@@ -67,6 +67,7 @@ export default function (space, options = {}) {
       let nodeLevel = indentationLevel(node)
 
       if (options && options.hierarchicalSelectors) {
+        console.log("Option hierarchicalSelectors has been deprecated and will be removed in 7.0") // eslint-disable-line no-console
```
Comment (jeddy3) [inline src/rules/indentation/index.js]: FYI, we have a special deprecation warning type that the formatters treat differently. Example of usage: https://github.com/stylelint/stylelint/blob/master/src/postcssPlugin.js#L60-L66

**Comment by jeddy3** [comment]
Good stuff. I think focusing on the deprecations is the only things left to do before we pack-up tools for our hols.

On the agenda is:

1. Coming to an agreement on https://github.com/stylelint/stylelint/issues/1308 - @davidtheclark when you have a mo, can you add your final thoughts to that issue?
2. We need to decide on how best to handle the deprecations. We can do that in this PR....

Previously we've had an alternative ready, and the deprecation messages would go something like "The 'old-option' option within 'rule-name' as be deprecated and will be moved in 7.0'. Use the 'new-option' option instead. See: http://stylelint.io/rule-name/#new-option"

This time we don't have:

1. a ready "new-option" option
2. a logical reference link

I've two suggestions. 

1. We create a brief "migrating-versions.md" doc within the user-guide. It outlines which options are being deprecated to create a more sustainable project (by either removing lesser-used features plugins, or removing ambiguity in rule names), and that we're looking to the community to fill the gap with plugins. We would link to this doc as the reference link. 
2. We then add a call of action to the deprecation warning to create a plugin. 

As such, the deprecation warning for this PR would read:

"The 'hierarchicalSelectors' option within 'indention' as be deprecated and will be moved in 7.0'. If you use this option please consider creating a `hierarchical-selectors-indentation` plugin for the community. See: http://stylelint.io/user-guides/migrating-versions.md#6to7"

**Comment by davidtheclark** [comment]
@jeddy3 Your idea for the deprecation warning sounds good to me. I will look into that other issue.

**Comment by ntwb** [comment]
> "...deprecated and will be moved in 7.0"

Typo, the above should read "... deprecated and will be removed in 7.0" i.e. `removed`, not `moved`

**Comment by jeddy3** [comment]
@davidtheclark As mentioned in https://github.com/stylelint/stylelint/pull/1345, I've reworded the deprecation warning and updated the reference link. Shall we merge?

**Comment by davidtheclark** [comment]
LGTM

**Pull request merged**